### ISSUE TITLE
update_attribute ignores autosave relations

### ISSUE THREAD

**Issue by Tietew** [rails/rails#25503]
### Steps to reproduce

``` ruby
require 'bundler/setup'
require 'active_record'
require 'sqlite3'
require 'minitest/autorun'

ActiveRecord::Base.establish_connection 'sqlite3::memory:'
ActiveRecord::Schema.define do
  create_table :people do |t|
    t.string :name
    t.timestamps null: false
  end
  create_table :emails do |t|
    t.belongs_to :person
    t.string :addr
  end
end

class Person < ActiveRecord::Base
  has_one :email, dependent: :destroy, autosave: true
  delegate :addr, :addr=, to: :email
end
class Email < ActiveRecord::Base
  belongs_to :person, touch: true
end

class BugTest < Minitest::Test
  def test_touch
    person = Person.create(name: 'Test')
    person.create_email(addr: 'test1@example.com')

    assert_equal 'test1@example.com', Person.find(person.id).addr # Pass

    # Expected to save associated Email model
    person.update_attribute(:addr, 'test2@example.com')

    assert_equal 'test2@example.com', Person.find(person.id).addr # FAIL HERE
  end
end
```
### Expected behavior (Rails 4.2.6)

The test passed.
update_attribute saves associated Email model.
### Actual behavior (Rails 5.0.0.rc2)

The test failed.
Nothing is saved.

Is it intentional?
### System configuration

**Rails version**: Rails 5.0.0.rc2

**Ruby version**: Ruby 2.3.1

**Comment by alexcameron89** [comment]
It looks like this behavior was added in 0fcd4cf5c26c470623eef9af72a134ef6ba1a701.

**Comment by maclover7** [comment]
Hmm, I'm not sure if this is supported behavior...

**Comment by rafaelfranca** [comment]
I'd say yes. It changed the behavior as consequence to fix an issue but it was never expected to remove support to this.

**Comment by uaihebert** [comment]
I've found the problem. 

The ``changed?`` method was not checking for an association update, but only for attributes update.

I need some ideas about the best way to check if an association have been updated. I needed to do some ifs because some tests were broken if I ran in group, but when I run individually it works. 

Thanks

**Comment by kamipo** [comment]
Fixed by #27780.
